ikuzo/service/x/ead/oaipmh.go

```diff
@@ -0,0 +1,161 @@
+package ead
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/delving/hub3/hub3/ead"
+	"github.com/delving/hub3/ikuzo/service/x/oaipmh"
+	"github.com/kiivihal/goharvest/oai"
+	"github.com/rs/zerolog/log"
+)
+
+type EADHarvester struct {
+	s     *Service
+	OrgID string
+}
+
+func NewEADHarvester(s *Service) (EADHarvester, error) {
+	if s == nil {
+		return EADHarvester{}, fmt.Errorf("nil value not allowed for ead.Service")
```

Comment:
If a nil value as ead.Service is not allowed. Wouldn't it be better to return nil as EADHarvester?

Reply:
We are not returning a pointer therefore it is better to return the empty struct.